ponyo/pipeline.py

```diff
@@ -127,28 +150,30 @@ def normalize_expression_data(base_dir, config_file, raw_input_data_file):
     dataset_name = params["dataset_name"]
 
     # Read data
-    rpkm_data = pd.read_table(raw_input_data_file, header=0, sep="\t", index_col=0)
+    data = pd.read_csv(raw_input_data_file, header=0, sep="\t", index_col=0)
 
     print(
-        "raw input dataset contains {} samples and {} genes".format(
-            rpkm_data.shape[0], rpkm_data.shape[1]
+        "input: dataset contains {} samples and {} genes".format(
+            data.shape[0], data.shape[1]
         )
     )
 
     # 0-1 normalize per gene
-    rnaseq_scaled_df = preprocessing.MinMaxScaler().fit_transform(rpkm_data)
-    rnaseq_scaled_df = pd.DataFrame(
-        rnaseq_scaled_df, columns=rpkm_data.columns, index=rpkm_data.index
-    ).T
+    data_scaled_df = preprocessing.MinMaxScaler().fit_transform(data)
+    data_scaled_df = pd.DataFrame(
+        data_scaled_df, columns=data.columns, index=data.index
+    )
 
-    # Save scaled data
-    normalized_data_file = os.path.join(
-        base_dir, dataset_name, "data", "input", "recount2_gene_normalized_data.tsv.xz"
+    print(
+        "Output: normalized dataset contains {} samples and {} genes".format(
+            data_scaled_df.shape[0], data_scaled_df.shape[1]
+        )
     )
 
-    rnaseq_scaled_df.to_csv(normalized_data_file, sep="\t", compression="xz")
+    # Save scaled data
+    data_scaled_df.to_csv(normalized_data_file, sep="\t", compression="xz")
 
-    del rnaseq_scaled_df
+    del data_scaled_df
     gc.collect()
```

Comment:
If this is at the end of a function shouldn't it automatically be collected by the garbage collector if you need memory? (This isn't a leading question where I think you're wrong, I legitimately don't understand how the python garbage collector works)

Reply:
You're right (https://rushter.com/blog/python-garbage-collector/)

These were added because I was having some memory issues that I couldn't figure out. Once I restarted my computer that seemed to fix things. So i think I will remove these.